tests/TestCase/Routing/Middleware/SecurityMiddlewareTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.5.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\Routing\Middleware;
+
+use Cake\Http\ServerRequestFactory;
+use Cake\Routing\Middleware\SecurityMiddleware;
+use Cake\TestSuite\TestCase;
+use Zend\Diactoros\Response;
+
+/**
+ * Test for SecurityMiddleware
+ */
+class SecurityMiddlewareTest extends TestCase
+{
+
+    /**
+     * Test adding the security headers
+     *
+     * @return void
+     */
+    public function testAddingSecurityHeaders()
+    {
+        $request = ServerRequestFactory::fromGlobals([
+            'REQUEST_URI' => '/',
+        ]);
+        $response = new Response();
+        $next = function ($req, $res) {
+            return $res;
+        };
+
+        $middleware = new SecurityMiddleware();
+        $middleware
+            ->setCrossDomainPolicy()
+            ->setReferrerPolicy()
+            ->setXFrameOptions()
+            ->setXssProtection()
+            ->noOpen()
```

Comment:
@lorenzo I named them intentionally this way because the set* methods have actually values you can set while the other two don't. You just "enable" them. Thought this makes the difference obvious.